calabash-cucumber/lib/calabash-cucumber/wait_helpers.rb

```diff
@@ -274,10 +274,9 @@ def wait_for_elements_do_not_exist(elements_arr, options={})
 
       # @!visibility private
       def wait_for_condition(options = {})
-        timeout = options[:timeout]
-        unless timeout && timeout > 0
-          timeout = 30
-        end
+        timeout = options[:timeout] || ENV['WAIT_TIMEOUT']
+        timeout = DEFAULT_OPTS[:timeout] unless timeout && timeout > 0
```

Comment:
I would really prefer it as `if !timeout || timeout =< 0` but this is a matter of preference :)